lib/Runtime/Library/InJavascript/Intl.js

```diff
@@ -608,20 +667,51 @@
         return toReturn;
     }());
 
+    // mappedDefaultLocale will get the default locale and update any deprecated
+    // collation/sort order values it may use
+    let __mappedDefaultLocale = undefined;
     var mappedDefaultLocale = function () {
-        var parts = platform.builtInRegexMatch(GetDefaultLocale(), /([^_]*)_?(.+)?/);
-        var locale = parts[1];
-        var collation = parts[2];
-        var availableBcpTags = reverseLocaleAcceptingCollationValues[locale];
+        if (__mappedDefaultLocale) {
+            return __mappedDefaultLocale;
+        }
 
-        if (collation === undefined || availableBcpTags === undefined) { return locale; }
+        let locale = undefined;
+        let collation = undefined;
+        if (isPlatformUsingICU) {
+            // ICU's getDefaultLocale() will return a valid BCP-47/RFC 5646 langtag
+            locale = GetDefaultLocale();
+            const match = platform.builtInRegexMatch(locale, /-u(-[^\-][^\-]?-[^\-]+)*-co-([^\-]+).*/);
+            if (match) {
+                collation = match[2];
+                locale = callInstanceFunc(StringInstanceReplace, locale, `-co-${match[2]}`, "");
+            }
+        } else {
+            // Windows' getDefaultLocale() will return a weird RFC4646 langtag
```

Comment:
>weird [](start = 57, length = 5)

nit (TODO): I think we should remove "weird" since we're just editorializing. It simply is an RFC4646 langtag, and weird here makes it sound like something about the tags are not quite RFC4646. Maybe maybe the distinction as: "RFC4646 (not RFC5646/BCP47) langtag"